src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -216,6 +207,58 @@ public void CheckForBrokerAccountInfoInAccountManager(IDictionary<string, string
             throw new MsalUiRequiredException(MsalError.NoAndroidBrokerAccountFound, MsalErrorMessage.NoAndroidBrokerAccountFound);
         }
 
+        private string GetBrokerAccounts(IDictionary<string, string> brokerPayload, Activity callerActivity)
+        {
+            Bundle getAccountsBundle = GetBrokerAccountBundle(brokerPayload);
+            getAccountsBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetAccounts);
+
+            //This operation will acquire all of the accounts in the account manager for the given client ID
+            var result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                BrokerConstants.AuthtokenType,
+                null,
+                getAccountsBundle,
+                null,
+                null,
+                GetPreferredLooper(callerActivity));
+
+            Bundle bundleResult = (Bundle)result?.Result;
+            return bundleResult?.GetString(BrokerConstants.BrokerAccounts);
+        }
+
+        /// <summary>
+        /// This method will acquire all of the accounts in the account manager that have an access token for the given client ID.
+        /// </summary>
+        public IEnumerable<IAccount> GetBrokerAccountsInAccountManager(IDictionary<string, string> brokerPayload)
+        {
+            var accounts = GetBrokerAccounts(brokerPayload, null);
+
+            if (string.IsNullOrEmpty(accounts))
+            {
+                _logger.Info("Android account manager didn't return any accounts.");
+            }
+
+            List<IAccount> brokerAccounts = new List<IAccount>();
+
+            if (!string.IsNullOrEmpty(accounts))
+            {
+                dynamic authResult = JArray.Parse(accounts);
+
+                foreach (JObject account in authResult)
+                {
+                    var accountData = account[BrokerResponseConst.Account];
```

Comment:
I think there's potential for some null refs here and line 250-253. Consider using `HasKey` to avoid this.

Reply:
did you see this comment?

---
In reply to: [385631543](https://github.com/azuread/microsoft-authentication-library-for-dotnet/pull/1668#discussion_r385631543) [](ancestors = 385631543)